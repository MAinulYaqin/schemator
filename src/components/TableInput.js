import React from "react";
import PropTypes from "prop-types";
import styled from "styled-components";
import { MdClose } from "react-icons/lib/md";

import "typeface-roboto";

const Container = styled.div`
  width: 100%;
  height: 36px;
  display: flex;
  align-items: center;
  justify-content: stretch;
  padding: 0px 4px;
  box-sizing: border-box;
`;

export const Select = styled.select`
  background: #333;
  color: white;
  font-family: Roboto;
  font-size: 11px;
  outline: none;
  width: 100%;
  height: 24px;
`;

export const Input = styled.input`
  width: 100%;
  height: 24px;
  color: white;
  background: #111;
  border: 0px;
  outline: none;
  font-family: Roboto;
  font-size: 12px;
  box-sizing: border-box;
  padding: 0px 8px;
`;

export const CloseButton = styled(MdClose)`
  color: #444;
  font-size: 16px;
  padding: 8px;
  cursor: pointer;
`;

const Column = styled.div`
  flex: 1;
  padding: 4px;
`;

const TableInput = ({
  name,
  type,
  onClickRemove,
  onChangeName,
  onChangeType
}) => {
  const types = [
    "BINARY",
    "BOOLEAN",
    "CHAR",
    "DATE",
    "DATE_TIME",
    "DATE_TIME_TZ",
    "DECIMAL",
    "DOUBLE",
    "ENUM",
    "FLOAT",
    "INTEGER",
    "IP_ADDRESS",
    "JSON",
    "JSONB",
    "MAC_ADDRESS",
    "MORPHS",
    "NULLABLE_MORPHS",
    "STRING",
    "TEXT",
    "TIME",
    "TIME_TZ",
    "TIMESTAMP",
    "TIMESTAMP_TZ",
    "UUID"
  ];

  return (
    <Container>
      <Column>
        <Input type="text" value={name} onChange={onChangeName} />
      </Column>
      <Column>
        <Select value={type} onChange={onChangeType}>
          {types.map((type, index) => (
            <option key={index} value={type}>
              {type}
            </option>
          ))}
        </Select>
      </Column>
      <CloseButton onClick={onClickRemove} />
    </Container>
  );
};

TableInput.propTypes = {
  name: PropTypes.string,
  type: PropTypes.string,
  onClickRemove: PropTypes.func,
  onChangeName: PropTypes.func,
  onChangeType: PropTypes.func
};

export default TableInput;
