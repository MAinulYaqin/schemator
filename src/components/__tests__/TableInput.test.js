import React from "react";
import { shallow } from "enzyme";
import toJSON from "enzyme-to-json";

import TableInput, { CloseButton, Input, Select } from "../TableInput";

describe("TableInput", () => {
  const setup = propOverrides => {
    const props = {
      name: "username",
      type: "STRING",
      onChangeName: jest.fn(),
      onChangeType: jest.fn(),
      onClickRemove: jest.fn(),
      ...propOverrides
    };

    const wrapper = shallow(<TableInput {...props} />);

    return {
      wrapper,
      props
    };
  };

  it("should renders default", () => {
    const { wrapper } = setup();

    expect(toJSON(wrapper)).toMatchSnapshot();
  });

  it("should renders with disabled selection", () => {
    const { wrapper } = setup({
      name: "user_id",
      type: "INTEGER"
    });

    expect(wrapper.find(Select).props().disabled).toEqual(true);
  });

  it("should calls 'onClickRemove' while Remove button was clicked", () => {
    const { wrapper, props } = setup();

    wrapper.find(CloseButton).simulate("click");

    expect(props.onClickRemove).toBeCalled();
  });

  it("should calls 'onChangeName' while field name was changed", () => {
    const { wrapper, props } = setup();
    const event = {
      target: {
        value: "id"
      }
    };

    wrapper.find(Input).simulate("change", event);

    expect(props.onChangeName).toBeCalledWith(event);
  });

  it("should calls 'onChangeType' while field type was changed", () => {
    const { wrapper, props } = setup();
    const event = {
      target: {
        value: "BIGINT"
      }
    };

    wrapper.find(Select).simulate("change", event);

    expect(props.onChangeType).toBeCalledWith(event);
  });
});
