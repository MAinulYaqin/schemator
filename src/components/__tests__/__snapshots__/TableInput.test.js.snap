// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableInput should renders default 1`] = `
<styled.div>
  <styled.div>
    <styled.input
      onChange={[Function]}
      type="text"
      value="username"
    />
  </styled.div>
  <styled.div>
    <styled.select
      disabled={false}
      onChange={[Function]}
      value="STRING"
    >
      <option
        key="0"
        value="BINARY"
      >
        BINARY
      </option>
      <option
        key="1"
        value="BOOLEAN"
      >
        BOOLEAN
      </option>
      <option
        key="2"
        value="CHAR"
      >
        CHAR
      </option>
      <option
        key="3"
        value="DATE"
      >
        DATE
      </option>
      <option
        key="4"
        value="DATE_TIME"
      >
        DATE_TIME
      </option>
      <option
        key="5"
        value="DATE_TIME_TZ"
      >
        DATE_TIME_TZ
      </option>
      <option
        key="6"
        value="DECIMAL"
      >
        DECIMAL
      </option>
      <option
        key="7"
        value="DOUBLE"
      >
        DOUBLE
      </option>
      <option
        key="8"
        value="ENUM"
      >
        ENUM
      </option>
      <option
        key="9"
        value="FLOAT"
      >
        FLOAT
      </option>
      <option
        key="10"
        value="INTEGER"
      >
        INTEGER
      </option>
      <option
        key="11"
        value="IP_ADDRESS"
      >
        IP_ADDRESS
      </option>
      <option
        key="12"
        value="JSON"
      >
        JSON
      </option>
      <option
        key="13"
        value="JSONB"
      >
        JSONB
      </option>
      <option
        key="14"
        value="MAC_ADDRESS"
      >
        MAC_ADDRESS
      </option>
      <option
        key="15"
        value="MORPHS"
      >
        MORPHS
      </option>
      <option
        key="16"
        value="NULLABLE_MORPHS"
      >
        NULLABLE_MORPHS
      </option>
      <option
        key="17"
        value="STRING"
      >
        STRING
      </option>
      <option
        key="18"
        value="TEXT"
      >
        TEXT
      </option>
      <option
        key="19"
        value="TIME"
      >
        TIME
      </option>
      <option
        key="20"
        value="TIME_TZ"
      >
        TIME_TZ
      </option>
      <option
        key="21"
        value="TIMESTAMP"
      >
        TIMESTAMP
      </option>
      <option
        key="22"
        value="TIMESTAMP_TZ"
      >
        TIMESTAMP_TZ
      </option>
      <option
        key="23"
        value="UUID"
      >
        UUID
      </option>
    </styled.select>
  </styled.div>
  <Styled(MdClose)
    onClick={[Function]}
  />
</styled.div>
`;
